sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedRleValuesReader.java

```diff
@@ -170,27 +170,28 @@ public int readInteger() {
    *  }
    */
   public void readBatch(
-      int total,
-      int offset,
+      ParquetReadState state,
       WritableColumnVector values,
-      int maxDefinitionLevel,
       VectorizedValuesReader valueReader,
       ParquetVectorUpdater updater) throws IOException {
```

Comment:
not related to this PR: is there an `IOException` to throw?

Reply:
good catch - I think there isn't `IOException` to throw here and we can just remove it.

Reply:
For this one, let's proceed separately if needed. There might be more places like this.